src/SourceLocator/Type/FileIteratorSourceLocator.php

```diff
@@ -26,11 +28,16 @@ class FileIteratorSourceLocator implements SourceLocator
     private $fileSystemIterator;
 
     /**
+     * @var null|Locator
+     */
+    private $locator;
+
+    /**
      * @param \Iterator|\SplFileInfo[] $fileInfoIterator note: only \SplFileInfo allowed in this iterator
      *
      * @throws InvalidFileInfo In case of iterator not contains only SplFileInfo
      */
-    public function __construct(\Iterator $fileInfoIterator)
+    public function __construct(\Iterator $fileInfoIterator, ?Locator $locator = null)
```

Comment:
IMO, the dependency should be the parser, and the dependency shouldn't be optional.